File: examples/data_structures.py
# Data structures example
def process_list():
    """Example of list operations"""
    numbers = [1, 2, 3, 4, 5]
    doubled = [x * 2 for x in numbers]
    
    result = []
    for num in doubled:
        if num > 5:
            result.append(num)
    
    return result

def process_dict():
    """Example of dictionary operations"""
    data = {
        "name": "John",
        "age": 30,
        "city": "New York"
    }
    
    # Add new entry
    data["country"] = "USA"
    
    # Process values
    for key, value in data.items():
        print(f"{key}: {value}")
    
    return data

def string_operations():
    """Example of string operations"""
    text = "Hello, World!"
    
    # Basic operations
    upper_text = text.upper()
    lower_text = text.lower()
    split_text = text.split(", ")
    
    # String formatting
    name = "Python"
    version = 3.9
    message = f"Welcome to {name} {version}!"
    
    return {
        "original": text,
        "upper": upper_text,
        "lower": lower_text,
        "split": split_text,
        "formatted": message
    }

def control_flow_example():
    """Example of control flow statements"""
    results = []
    
    # For loop with range
    for i in range(10):
        if i % 2 == 0:
            results.append(f"Even: {i}")
        else:
            results.append(f"Odd: {i}")
    
    # While loop
    count = 0
    while count < 5:
        results.append(f"Count: {count}")
        count += 1
    
    return results

def error_handling_example():
    """Example of error handling"""
    try:
        result = 10 / 0
    except ZeroDivisionError as e:
        print(f"Error: {e}")
        result = None
    except Exception as e:
        print(f"Unexpected error: {e}")
        result = None
    finally:
        print("Cleanup code here")
    
    return result

def main():
    """Main function"""
    print("List processing result:", process_list())
    print("\nDictionary processing:")
    process_dict()
    print("\nString operations:", string_operations())
    print("\nControl flow example:", control_flow_example())
    print("\nError handling example:", error_handling_example())

if __name__ == "__main__":
    main()

File: examples/cpp_output/data_structures.cpp
#include <iostream>
#include <string>
#include <vector>
#include <memory>
#include <cmath>

class Calculator;
class ScientificCalculator;

class Calculator {
public:
    Calculator();
    
    auto add(auto a, auto b);
    auto subtract(auto a, auto b);
    auto multiply(auto a, auto b);
    auto divide(auto a, auto b);
    auto get_result();
    auto get_history();
    auto clear();
};

class ScientificCalculator : public Calculator {
public:
    ScientificCalculator();
    
    auto power(auto base, auto exponent);
    auto square_root(auto n);
    auto store_memory();
    auto recall_memory();
};

auto main();

auto __init__(auto self);

auto add(auto self, auto a, auto b);

auto subtract(auto self, auto a, auto b);

auto multiply(auto self, auto a, auto b);

auto divide(auto self, auto a, auto b);

auto get_result(auto self);

auto get_history(auto self);

auto clear(auto self);

auto __init__(auto self);

auto power(auto self, auto base, auto exponent);

auto square_root(auto self, auto n);

auto store_memory(auto self);

auto recall_memory(auto self);

auto process_list();

auto process_dict();

auto string_operations();

auto control_flow_example();

auto error_handling_example();

auto main();

auto main() {
    "Main function";
    std::cout << "List processing result:" << " " << process_list() << std::endl;
    std::cout << "
Dictionary processing:" << std::endl;
    process_dict();
    std::cout << "
String operations:" << " " << string_operations() << std::endl;
    std::cout << "
Control flow example:" << " " << control_flow_example() << std::endl;
    std::cout << "
Error handling example:" << " " << error_handling_example() << std::endl;
}

auto process_list() {
    "Example of list operations";
    auto numbers = {1, 2, 3, 4, 5};
    auto doubled = /* TODO: ListComp */;
    auto result = {};
    // TODO: Complex for loop
        if (num > 5) {
            result.push_back(num);
        }
    }
    return result;
}

auto process_dict() {
    "Example of dictionary operations";
    auto data = /* TODO: Dict */;
    // TODO: Complex assignment
    // TODO: Complex for loop
        std::cout << /* TODO: JoinedStr */ << std::endl;
    }
    return data;
}

auto string_operations() {
    "Example of string operations";
    auto text = "Hello, World!";
    auto upper_text = text.upper();
    auto lower_text = text.lower();
    auto split_text = text.split(", ");
    auto name = "Python";
    auto version = 3.9;
    auto message = /* TODO: JoinedStr */;
    return /* TODO: Dict */;
}

auto control_flow_example() {
    "Example of control flow statements";
    auto results = {};
    for (int i = 0; i < 10; ++i) {
        if ((i % 2) == 0) {
            results.push_back(/* TODO: JoinedStr */);
        }
        else {
            results.push_back(/* TODO: JoinedStr */);
        }
    }
    auto count = 0;
    while (count < 5) {
        results.push_back(/* TODO: JoinedStr */);
        // TODO: Implement AugAssign
    }
    return results;
}

auto error_handling_example() {
    "Example of error handling";
    // TODO: Implement Try
    return result;
}

auto main() {
    "Main function";
    std::cout << "List processing result:" << " " << process_list() << std::endl;
    std::cout << "
Dictionary processing:" << std::endl;
    process_dict();
    std::cout << "
String operations:" << " " << string_operations() << std::endl;
    std::cout << "
Control flow example:" << " " << control_flow_example() << std::endl;
    std::cout << "
Error handling example:" << " " << error_handling_example() << std::endl;
}